main.js

```diff
@@ -29,18 +45,62 @@ var data = {
   started: false
 };
 
-var moveCar;
+var moveCarRight;
+var moveCarLeft;
+var moveCarUp;
+var moveCarDown;
 
-function move(){
-  $car.style.top = data.location.y;
+function move(direction){
   data.started = true;
-  moveCar = setInterval(function(){
-    data.location.x++;
-    $car.style.left = data.location.x.toString().concat('px');
-  }, 16)
+  if(direction === 'right'){
+    clearInterval(moveCarLeft);
+    clearInterval(moveCarDown);
+    clearInterval(moveCarUp);
+    moveCarRight = setInterval(function () {
+      data.location.x+=10;
+      $car.style.top = data.location.y.toString().concat('px');
+      $car.style.left = data.location.x.toString().concat('px');
+    }, 16)
+  }
+
+  if(direction === 'left'){
+    clearInterval(moveCarRight);
+    clearInterval(moveCarDown);
+    clearInterval(moveCarUp);
+    moveCarLeft = setInterval(function () {
+      data.location.x-=10;
+      $car.style.top = data.location.y.toString().concat('px');
+      $car.style.left = data.location.x.toString().concat('px');
+    }, 16)
+  }
+
+  if(direction === 'up'){
+    clearInterval(moveCarLeft);
+    clearInterval(moveCarRight);
+    clearInterval(moveCarDown);
+    moveCarUp = setInterval(function () {
+      data.location.y-=10;
+      $car.style.top = data.location.y.toString().concat('px');
+      $car.style.left = data.location.x.toString().concat('px');
+    }, 16)
+  }
+
+  if(direction === 'down'){
+    clearInterval(moveCarLeft);
+    clearInterval(moveCarRight);
+    clearInterval(moveCarUp);
+    moveCarDown = setInterval(function () {
+      data.location.y+=10;
+      $car.style.top = data.location.y.toString().concat('px');
+      $car.style.left = data.location.x.toString().concat('px');
+    }, 16)
+  }
 }
 
 function stop(){
-  clearInterval(moveCar);
+  clearInterval(moveCarLeft);
+  clearInterval(moveCarRight);
+  clearInterval(moveCarUp);
+  clearInterval(moveCarDown);
   data.started = false;
```

Comment:
You need one interval.
The constantly updating interval will be the one moving the car.